test/unit/core/common.js

```diff
@@ -11,30 +11,39 @@ class ReplSetFixture {
     this.electionIds = [new ObjectId(), new ObjectId()];
   }
 
+  uri(dbName) {
+    return `mongodb://${this.primaryServer.uri()},${this.firstSecondaryServer.uri()},${this.secondSecondaryServer.uri()}/${dbName ||
+      'test'}?replicaSet=rs`;
+  }
+
   setup(options) {
     options = options || {};
     const ismaster = options.ismaster ? options.ismaster : mock.DEFAULT_ISMASTER_36;
 
-    return Promise.all([mock.createServer(), mock.createServer(), mock.createServer()]).then(
-      servers => {
-        this.servers = servers;
-        this.primaryServer = servers[0];
-        this.firstSecondaryServer = servers[1];
-        this.arbiterServer = servers[2];
-
-        this.defaultFields = Object.assign({}, ismaster, {
-          __nodejs_mock_server__: true,
-          setName: 'rs',
-          setVersion: 1,
-          electionId: this.electionIds[0],
-          hosts: this.servers.map(server => server.uri()),
-          arbiters: [this.arbiterServer.uri()]
-        });
-
-        this.defineReplSetStates();
-        this.configureMessageHandlers();
-      }
-    );
+    return Promise.all([
+      mock.createServer(),
+      mock.createServer(),
+      mock.createServer(),
+      mock.createServer()
+    ]).then(servers => {
+      this.servers = servers;
+      this.primaryServer = servers[0];
+      this.firstSecondaryServer = servers[1];
+      this.secondSecondaryServer = servers[2];
+      this.arbiterServer = servers[3];
+
+      this.defaultFields = Object.assign({}, ismaster, {
+        __nodejs_mock_server__: true,
+        setName: 'rs',
+        setVersion: 1,
+        electionId: this.electionIds[0],
+        hosts: this.servers.map(server => server.uri()),
+        arbiters: [this.arbiterServer.uri()]
+      });
+
+      this.defineReplSetStates();
+      // this.configureMessageHandlers();
```

Comment:
Should this line be commented out? Seems there are some issues in CI.

Reply:
Whoops good catch, let's see if the commit I just pushed fixes it. I made the change on the 4.0 port already to add if statements to gate these extra setup calls.